README.md

```diff
@@ -4,111 +4,209 @@ An API for validating survey schemas.
 
 ## Setup
 
-It is recommended that you use [Pyenv](https://github.com/pyenv/pyenv) to manage your Python installations.
+In order to run locally you'll need Node.js, poetry and python installed. It's recommended that python is installed via pyenv but pyenv is optional.
 
-### Install Poetry
+### Install NVM and pyenv
+
+NVM and pyenv will manage your versions of Node and Python and these commands will install the required versions of them which will be read from `.nvmrc` and `.python-version`.
+
+``` shell
+brew install nvm pyenv
+nvm use
+pyenv install
+```
+
+If you get a message in the command line after running `nvm use` that the version of node specified in the `.nvmrc` file isn't installed, just follow the commands to install it.
+
+e.g.
+``` shell
+nvm install v22.15.0
+```
+
+### Install JS dependencies
+
+``` shell
+npm install
 ```
+
+### Install Poetry and Python dependencies
+
+``` shell
 curl -sSL https://install.python-poetry.org | python3 - --version 2.1.2
 poetry install
 ```
 
-## Running 
+## Running locally
+
+To run the app:
+``` shell
+make run
+```
+
+Validator runs on two ports, `5001` is the main validator app and `5002` is the Ajv validator.
+
+### Validator
 
-The `AJV_VALIDATOR_URL` defaults to `http://localhost:5002/validate`.
+Validator runs on `http://localhost:5001/validate`.
+
+If you need to change the port you can change the port variable in the Uvicorn settings in api.py:
+``` python
+uvicorn.run("api:app", workers=20, port=5001, reload=False)
+```
+If you want to run the app locally using multiple server workers you can also set reload to "False" here too.
```

Comment:
I think the `reload` flag will allow the service to restart if you make a change to the code if it's set to `True`?